### PR TITLE
Refactor automated tests, add pre-commit and CI linter checks.

### PR DESCRIPTION
**Pull request by yashtewari**
Pre-commit checks can now be installed with
```
just linter-setup
```

### CODE REVIEW

**Review by oren-zohar — CHANGES_REQUESTED** [review]
Could you please summarize the checks we are going to ignore and the reasoning behind each one?

**Comment by yashtewari** [comment]
> Could you please summarize the checks we are going to ignore and the reasoning behind each one?

@oren-zohar I've added the explanations within the code itself, and put the pytlinrc changes in a separate commit so it can be reviewed easily.

File: tests/commonlib/io_utils.py [hunk 1]
```diff
@@ -2,19 +2,23 @@
...
 from pathlib import Path
 
+import yaml
+from munch import Munch, munchify
 
-def get_events_from_index(elastic_client, index_name: str, rule_tag: str, time_after: datetime) -> list[Munch]:
+
+def get_events_from_index(
+    elastic_client, index_name: str, rule_tag: str, time_after: datetime
+) -> list[Munch]:
```
Comment (oren-zohar) [inline tests/commonlib/io_utils.py:21]: please fix line breaks

**Review by oren-zohar — CHANGES_REQUESTED** [review]
many line breaks need a fix, please ping me when this is ready for a review

**Review by oren-zohar — CHANGES_REQUESTED** [review]
I suggest using [black](https://github.com/psf/black) formatter to handle the line breaks automatically, or [yapf](https://github.com/google/yapf) (see `split_arguments_when_comma_terminated`)

File: scripts/git-hooks/pre-commit [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+source tests/venv/bin/activate
+
+echo "Checking with flake8..."
+flake8 --ignore="W503,W504,E203,E501" tests --include "tests/" --exclude "tests/venv"
+
+echo ""
```
[the remark this answers is not in the document]
Comment (yashtewari) [reply scripts/git-hooks/pre-commit:8]: To add a newline between the two reports.

File: scripts/linter_setup.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (yashtewari) [reply scripts/linter_setup.sh:1]: The shebang?

File: scripts/git-hooks/pre-commit [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+source tests/venv/bin/activate
+
+echo "Checking with flake8..."
+flake8 --ignore="W503,W504,E203,E501" tests --include "tests/" --exclude "tests/venv"
```
[the remark this answers is not in the document]
Comment (yashtewari) [reply scripts/git-hooks/pre-commit]: Have moved flake8 configuration to `setup.cfg` and added comments explaining the need to ignore stuff.

**Comment by yashtewari** [comment]
> I suggest using [black](https://github.com/psf/black) 

This is a great option; I especially like that it leaves no scope for any debate on style by enforcing its own style that can't be changed across projects. I've added it to the pre-commit checks. Unfortunately, it doesn't agree with our one-per-line function parameters from before, so it may be time to let go of them.

**Comment by yashtewari** [comment]
Switched to using the [pre-commit](https://pre-commit.com) package. Fixed the multi-line parameter request.

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**